sql/mysql/tidb.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2021-present The Atlas Authors. All rights reserved.
+// This source code is licensed under the Apache 2.0 license found
+// in the LICENSE file in the root directory of this source tree.
+
+package mysql
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"ariga.io/atlas/sql/internal/sqlx"
+	"ariga.io/atlas/sql/schema"
+)
+
+type tinspect struct {
+	inspect
+}
+
+func (i *tinspect) InspectSchema(ctx context.Context, name string, opts *schema.InspectOptions) (*schema.Schema, error) {
+	s, err := i.inspect.InspectSchema(ctx, name, opts)
+	if err != nil {
+		return nil, err
+	}
+	return i.patchSchema(ctx, s)
+}
+
+func (i *tinspect) InspectRealm(ctx context.Context, opts *schema.InspectRealmOption) (*schema.Realm, error) {
+	r, err := i.inspect.InspectRealm(ctx, opts)
+	if err != nil {
+		return nil, err
+	}
+	for _, s := range r.Schemas {
+		if _, err := i.patchSchema(ctx, s); err != nil {
+			return nil, err
+		}
+	}
+	return r, nil
+}
+
+func (i *tinspect) patchSchema(ctx context.Context, s *schema.Schema) (*schema.Schema, error) {
+	for _, t := range s.Tables {
+		var createStmt CreateStmt
+		if ok := sqlx.Has(t.Attrs, &createStmt); !ok {
+			if err := i.createStmt(ctx, t); err != nil {
+				return nil, err
+			}
+		}
+		if err := i.setCollate(t); err != nil {
+			return nil, err
+		}
+		if err := i.setFKs(s, t); err != nil {
+			return nil, err
+		}
+	}
+	return s, nil
+}
+
+// e.g CONSTRAINT "" FOREIGN KEY ("foo_id") REFERENCES "foo" ("id")
+var reFK = regexp.MustCompile("(?i)CONSTRAINT\\s+[\"`]*(\\w+)[\"`]*\\s+FOREIGN\\s+KEY\\s*\\(([,\"` \\w]+)\\)\\s+REFERENCES\\s+[\"`]*(\\w+)[\"`]*\\s*\\(([,\"` \\w]+)\\)")
+
+func (i *tinspect) setFKs(s *schema.Schema, t *schema.Table) error {
+	var c CreateStmt
+	if !sqlx.Has(t.Attrs, &c) {
+		return fmt.Errorf("missing CREATE TABLE statment in attribuets for %q", t.Name)
+	}
+	for _, m := range reFK.FindAllStringSubmatch(c.S, -1) {
+		if len(m) != 5 {
+			return fmt.Errorf("unexpected number of matches for a table constraint: %q", m)
+		}
+		ctName, clmns, refTableName, refClmns := m[1], m[2], m[3], m[4]
+		fk := &schema.ForeignKey{
+			Symbol: ctName,
+			Table:  t,
+			// There is no support in TiDB for FKs so inherently there are no actions on update/delete.
+			OnUpdate: schema.NoAction,
+			OnDelete: schema.NoAction,
```

Comment:
Are you sure it won't create a diff in case the referential action is not "NO ACTION"?

Reply:
no, but if it does I'll find out soon enough with the integration tests.  
Why would there be any action other than that? they don't support this constraint anyway :)